src/main/java/net/gecore/streamerendpoints/service/twitch/SharedAPI.java

```diff
@@ -0,0 +1,73 @@
+package net.gecore.streamerendpoints.service.twitch;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SharedAPI {
+
+    //private final CommonRateLimitService rateLimitService;
+
+    private final Logger LOGGER = LoggerFactory.getLogger(SharedAPI.class);
+
+    @Autowired

```

Comment:
This constructor + annotation is unnecessary. As you are not trying to autowire anything in you can leave it out.